core/shared/src/main/scala-2/zio/internal/macros/LayerMacroUtils.scala

```diff
@@ -102,12 +102,12 @@ private[zio] trait LayerMacroUtils {
       z = reify(ZLayer.unit),
       value = memoMap,
       composeH = (lhs, rhs) => c.Expr(q"$lhs ++ $rhs"),
-      composeV = (lhs, rhs) => c.Expr(q"$compose($lhs, $rhs)")
+      composeV = (lhs, rhs) => c.Expr(q"$compose($lhs, $rhs)($trace)")
     )
 
     c.Expr(q"""
-    implicit val $trace: ${typeOf[Trace]} = ${reify(Tracer)}.newTrace
-    def $compose[R1, E, O1, O2](lhs: $layerSym[R1, E, O1], rhs: $layerSym[O1, E, O2]) = lhs.to(rhs)
+    val $trace: ${typeOf[Trace]} = ${reify(Tracer)}.newTrace
```

Comment:
We need to also pass it explicitly to `ZIO.environment` but when I tried, it's giving me a weird error:
```
[error] /Users/georgy/work/zio/zio/core-tests/shared/src/test/scala/zio/autowire/AutoWireSpec.scala:215:43: exception during macro expansion: 
[error] java.lang.NullPointerException: Cannot invoke "scala.reflect.api.Types$TypeApi.$less$colon$less(scala.reflect.api.Types$TypeApi)" because the return value of "scala.reflect.api.Trees$TreeApi.tpe()" is null
[error]         at zio.internal.macros.CleanCodePrinter$.$anonfun$cleanImplicits$1(CleanCodePrinter.scala:132)
[error]         at zio.internal.macros.CleanCodePrinter$.$anonfun$cleanImplicits$1$adapted(CleanCodePrinter.scala:132)
[error]         at scala.collection.immutable.List.exists(List.scala:396)
[error]         at zio.internal.macros.CleanCodePrinter$.loop$1(CleanCodePrinter.scala:132)
[error]         at zio.internal.macros.CleanCodePrinter$.cleanImplicits(CleanCodePrinter.scala:148)
[error]         at zio.internal.macros.CleanCodePrinter$.show(CleanCodePrinter.scala:18)
[error]         at zio.internal.macros.LayerMacroUtils.$anonfun$constructLayer$5(LayerMacroUtils.scala:70)
[error]         at zio.internal.macros.LayerTree.$anonfun$map$1(LayerTree.scala:24)
[error]         at zio.internal.macros.LayerTree.fold(LayerTree.scala:16)
[error]         at zio.internal.macros.LayerTree.fold(LayerTree.scala:18)
[error]         at zio.internal.macros.LayerTree.fold(LayerTree.scala:20)
[error]         at zio.internal.macros.LayerTree.map(LayerTree.scala:24)
[error]         at zio.internal.macros.LayerBuilder.warnUnused(LayerBuilder.scala:154)
[error]         at zio.internal.macros.LayerBuilder.build(LayerBuilder.scala:105)
[error]         at zio.internal.macros.LayerMacroUtils.constructLayer(LayerMacroUtils.scala:76)
[error]         at zio.internal.macros.LayerMacroUtils.constructLayer$(LayerMacroUtils.scala:41)
[error]         at zio.internal.macros.ZLayerMakeMacros.constructLayer(ZLayerMakeMacros.scala:8)
[error]         at zio.internal.macros.ZLayerMakeMacros.makeSomeImpl(ZLayerMakeMacros.scala:34)
[error]               ZLayer.makeSome[Double, Int](intLayer, stringLayer)
[error]                                           ^
[error] one error found
[error] (coreTestsJVM / Test / compileIncremental) Compilation failed
[error] Total time: 22 s, completed 13 Jun 2024, 09:45:44
```